### ISSUE TITLE
Jupyter doesn't work in creator update

### ISSUE THREAD

**Issue by Lynntee** [Microsoft/WSL#1857]
* A brief description

I update to 15063.13 1703 today. Then I run `sudo do-release-upgrade` to upgrade WSL to ubuntu 16.04. After that, I found Jupyter doesn't work. 

When I run `jupyter qtconsole`, kernel died and try to restart, again and again. 

When I run `jupyter notebook --no-browser`, it shows there is 0 active kernels.

* Expected results

Before I update to 15063, I can use jupyter following #185 .But today ,it doesn't work for me.

When I run

```
sudo add-apt-repository ppa:aseering/wsl
sudo apt-get update
sudo apt-get install libzmq3
```
I failed during `apt update`, it shows
`W: The repository 'http://ppa.launchpad.net/aseering/wsl/ubuntu xenial Release' does not have a Release file.`

Then `apt-get install libzmq3` return `E: Unable to locate package libzmq3`.

I have no idea about that. What can I do to fix?

* Your Windows build number

15063.13  1703

* Required packages and commands to install

I am using `anaconda`.

Thanks!

**Comment by aseering** [comment]
Hi @Lynntee -- thanks for posting!

The last time I checked, jupyter worked fine on Creator's.  I have not had any trouble with it.

The PPA above does not work with Creators because it is no longer needed.  It fixed one specific bug, and that bug has been worked around by code that's present in Creator's.

Did jupyter work for you after you upgraded to Creators but before you upgraded to 16.04?  If so, I'm guessing something about Ubuntu's upgrade script broke your system.  (If you had the PPA above installed, it should have errored out and forced you to remove the PPA and the package because, as you've seen, the PPA is not compatible with Ubuntu 16.04.)  If that's the case, you may need to update or reinstall Anaconda, or possibly more, in order to get back to a clean slate.

**Comment by Lynntee** [comment]
@aseering 

After I upgraded to Creators, I upgraded to 16.04 immediately, so I don't know if it works that time.

I try to reinstall the newest version of Anaconda, it's same as before.

**Comment by aseering** [comment]
@Lynntee -- hm...  Did you upgrade Anaconda, or did you upgrade the packages that Anaconda had installed?  These are separate steps; reinstalling Anaconda doesn't necessarily update its packages.

**Comment by Lynntee** [comment]
@aseering I just download the newest version of Anaconda and install it. Now in my WSL, the version of Anaconda is 4.3.1, I use 4.1.1 before, maybe it does matter? Or maybe I can try to reinstall WSL?

**Comment by karlti** [comment]
FYI
i am on Win10 build 16170.
installed pip3 and did a "`pip install jupyter`".
and then started jupyter:

```
(ass1) karl@CALYPSO:~/cs231n/ass1$ jupyter notebook                               
(ass1) karl@CALYPSO:~/cs231n/ass1$ jupyter notebook                               
[I 15:44:21.203 NotebookApp] Writing notebook server cookie secret to /home/karl.....                             
[I 15:44:21.266 NotebookApp] Serving notebooks from local directory: /home/karl/cs231n/ass1                                                                         
[I 15:44:21.266 NotebookApp] 0 active kernels                                     
[I 15:44:21.266 NotebookApp] The Jupyter Notebook is running at: http://localhost:8888/?token=47d....                      
[I 15:44:21.266 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).                                              
[W 15:44:21.271 NotebookApp] No web browser found: could not locate runnable browser.                                                                               
[C 15:44:21.272 NotebookApp]                                                      
                                                                                  
    Copy/paste this URL into your browser when you connect for the first time,    
    to login with a token:                                                        
        http://localhost:8888/?token=47d50f8abf643641e....

```
when i then started Chrome on Windows and pasted the above URL **Bingo!**

Jupyter started & worked

**Comment by Lynntee** [comment]
@karlti Thank you for your reply. But I'm sorry to say, I copied the url and open it in chrome on windows, the page can not be opened, showing no connection.

**Comment by Lynntee** [comment]
@aseering Hey, I think I have found what the problem is. It seems that the problem is any port on WSL doesn't work. See #1498 . 

I found that when I run `jupyter notebook` and copy / paste the url into browser on windows, the error was `ERR_CONNECTION_REFUSED`, it's weird. 

So I run a django app in WSL, but I can't visit localhost:8000 with the same error.
Farther more, I want to determine whether the problem is just happen between WSL and windows, so I run `jupyter notebook` on windows, and I can visit the url on WSL.

What's more, when I use emacs to write python, the `anaconda-mode` which opens a port doesn't work at all.

I have tried to close firewall and AV on windows, it doesn't work.
